File: 1-easy/problems/1935-maximum-number-of-words-you-can-type/solution.py
class Solution:
    def canBeTypedWords(self, text, brokenLetters):
        """
        :type text: str
        :type brokenLetters: str
        :rtype: int
        """
        lookup = set(brokenLetters)
        result, broken = 0, False
        for c in text:
            if c == ' ':
                result += int(broken == False)
                broken = False
            elif c in lookup:
                broken = True
        return result + int(broken == False)

File: 1-easy/problems/1935-maximum-number-of-words-you-can-type/solution.ts
function canBeTypedWords(text: any, brokenLetters: any): boolean | number | string | any {
  """
  :type text: str
  :type brokenLetters: str
  :rtype: int
  """
  lookup = set(brokenLetters)
  result, broken = 0, false
  for (c in text) {
      if (c == ' ') {
          result += int(broken == false)
          broken = false
      } else if (c in lookup:
          broken = true
  return result + int(broken == false);
}

export { canBeTypedWords };
